### DISCUSSION TITLE
optimize RF for spearman correlation?

### DISCUSSION THREAD

**Discussion by S90pngprgm** [kogalur/randomForestSRC#213] [Q&A]
I am trying to find a prediction so that the predicted responses and the true responses achieve the best spearman correlation coefficient. I understand than randomforest is not optimized in this way.

I have tried quantile regression with quantile of .5. The spearman correlation that I can achieve with quantial regression at q=.5 is even less than a feature of best spearman correlation with the response variable. Is this an expected result?

What is the best method to predict the response variable so that its spearman correlation with the true response is maximized?

**Comment by ishwaran** [comment]
The best way would be to devise your own custom splitting rule that splits the node using the feature and split value that maximizes the Spearman correlation.   The package allows users to develop their own splitting rules.

See the discussion about custom splitting  rules here:

https://github.com/kogalur/randomForestSRC/discussions/197#discussioncomment-2277679

**Reply by S90pngprgm** [reply]
Thanks. I am not sure I understand how it works. The Spearman correlation is to be maximized for all the testing data. But during splits, only data without a node are available. This seems to be a disconnect between the two. Could you please explain how the splitting rules on the trainding data can be made to optimize the Spearman correlation of the test data?

**Comment by ishwaran** [comment]
Well you need to think about this and come up with something clever!

But just look at the MSE splitting rule.  It splits the parent so that there is low variance in each daughter.  This subsequently leads to good out of sample performance, but obviously we don't use the test data to ensure that this happens.  That is, we don't use test data to do the splitting.

We also have an AUC splitting rule.  The AUC might seem like something that couldn't be used for splitting, but with some clever thinking, it is possible to implement something that is designed to improve AUC performance.
